src/status_im/contexts/wallet/effects.cljs

```diff
@@ -17,3 +21,66 @@
       {:MnemonicPhrase phrase
        :paths          paths}
       on-success))))
+
+(defn validate-mnemonic
+  [mnemonic]
+  (-> mnemonic
+      (security/safe-unmask-data)
+      (native-module/validate-mnemonic)
+      (promesa/then (fn [result]
+                      (let [{:keys [error keyUID]} (transforms/json->clj result)]
+                        (if (seq error)
+                          (promesa/rejected error)
+                          {:key-uid keyUID}))))))
+
+(rf/reg-fx
+ :multiaccount/validate-mnemonic
+ (fn [[mnemonic on-success on-error]]
+   (-> (validate-mnemonic mnemonic)
+       (promesa/then (fn [{:keys [key-uid]}]
+                       (when (fn? on-success)
```

Comment:
If I remember correctly, somebody wrote a little abstraction to help define effects that can take either callbacks or event vectors (which follow the standard used in `json-rpc/call` to conj into event vectors, and same convention used by https://github.com/day8/re-frame-http-fx).

Was it @clauxx maybe? Well, even if such abstraction doesn't exist yet, it's an idea for the future because it would make our effects more consistent and it would help us test events because we wouldn't be passing anonymous functions.

Reply:
@ilmotta yeah it was removed in this [commit](https://github.com/status-im/status-mobile/pull/18756/commits/52ee754bc58a2bb7a1d13f37f0d16d03b2a625e4)